ansible/roles/baselayout/tasks/ccache.yml

```diff
@@ -23,10 +23,17 @@
   register: new_archive
 
 - name: "ccache : prepare for compilation"
+  when: not os|startswith("smartos")
   shell: ./configure -q
   args:
     chdir: "/tmp/ccache-{{ ccache_latest }}"
 
+- name: "ccache : prepare for compilation - smartos"
+  when: os|startswith("smartos")
+  shell: bash ./configure CPPFLAGS=-std=c99
```

Comment:
I think I remember this but may have resorted to an older version of ccache to get around it. Good change, thanks for figuring it out!